scripts/create_tflm_arduino.py

```diff
@@ -0,0 +1,504 @@
+
+  def _GenerateBasePathsRelative(
+      self, generate_transform_paths: bool) -> List[Tuple[Path, Path]]:
+
+    # generate set of all files from base directory
+    all_files = self.base_dir.glob("**/*")
+    # filter out directories
+    all_files = filter(lambda p: not p.is_dir(), all_files)
+    special_path_dict = dict(self.special_base_list)
+
+    # generate relative source/destination pairs
+    relative_path_pairs: List[Tuple[Path, Path]] = []
+    for file in all_files:
+      src_path = Path(file).relative_to(self.base_dir)
+      # check if in manifest special base or remove list
+      if special_path_dict.get(src_path) is not None:
+        continue
+      elif src_path in self.remove_list:
+        continue
+      # tensorflow/ and third_party/ will be subdirectories of src/
+      if src_path.parts[0] in ["third_party", "tensorflow"]:
+        dst_path = "src" / src_path
+      else:
+        dst_path = src_path
+
+      # check for .cc to .cpp rename
+      # future: apply suffixes_mapping= instead
+      if dst_path.suffix == ".cc":
+        dst_path = dst_path.with_suffix(".cpp")
+
+      # add new tuple(src,dst) to list
+      relative_path_pairs.append((src_path, dst_path))
+
+    # add all manifest special base paths
+    for paths in self.special_base_list:
+      from_path = paths[0]
+      to_path = paths[1]
+      # future: apply suffixes_mapping=
+      relative_path_pairs.append((from_path, to_path))
+
+    # filter for (non)transformable
+    # future: use transform_suffixes=
+    xform_suffixes = [".c", ".cc", ".cpp", ".h", ".ino"]
+    relative_path_pairs = list(
+        filter(
+            lambda pair: generate_transform_paths ==
+            (pair[0].suffix in xform_suffixes), relative_path_pairs))
+
+    return relative_path_pairs
+
+  def _GenerateRepoPathsRelative(
+      self, generate_transform_paths: bool) -> List[Tuple[Path, Path]]:
+
+    relative_path_pairs: List[Tuple[Path, Path]] = []
+    if generate_transform_paths:
+      # future: download (url=) sections using transform_suffixes= list
+      # future: apply suffixes_mapping=
+      # for now just return empty list
+      pass
+    else:
+      # future: download (url=) sections not in transform_suffixes= list
+      for path in self.add_list:
+        # future: apply suffixes_mapping=
+        relative_path_pairs.append((path, path))
+      for paths in self.special_repo_list:
+        from_path = paths[0]
+        to_path = paths[1]
+        # future: apply suffixes_mapping=
+        relative_path_pairs.append((from_path, to_path))
+
+    return relative_path_pairs
+
+  def _RemoveEmptyDirectories(self) -> None:
+    paths = self.output_dir.glob("**")
+    _remove_empty_directories(paths,
+                              root=self.output_dir,
+                              is_dry_run=self.is_dry_run)
+
+  #
+  # public methods
+  #
+
+  def CreateZip(self) -> None:
+    """
+    Execute all steps to create TFLM Arduino ZIP file
+    """
```

Comment:
Let's discuss what we want the script output to be. I would prefer this script to change the repository in-place with updated files from tflite-micro unless that would become quite a bit more complex.